zanata-war/src/main/java/org/zanata/events/DocStatsEvent.java

```diff
@@ -0,0 +1,58 @@
+package org.zanata.events;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.zanata.common.ContentState;
+
+import lombok.AllArgsConstructor;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+
+import javax.annotation.Nullable;
+
+/**
+ * @author Alex Eng <a href="mailto:aeng@redhat.com">aeng@redhat.com</a>
+ */
+@AllArgsConstructor
+@Getter
+@EqualsAndHashCode
+public class DocStatsEvent {
+    private final DocumentLocaleKey key;
+
+    private final Long projectVersionId;
+
+    /**
+     * Updated content states with word counts
+     */
+    private final Map<ContentState, Integer> contentStates;
+
+    private final Long lastModifiedTargetId;
+
+    public static Map<ContentState, Integer> updateContentState(
+        @Nullable Map<ContentState, Integer> contentStates, ContentState newState,
+        ContentState previousState, Long longWordCount) {
+
+        Map<ContentState, Integer> newContentStates =
+            contentStates == null ? new HashMap<ContentState, Integer>() :
+                contentStates;
+        int wordCount = longWordCount == null ? 0 : longWordCount.intValue();
+
+        Integer previousStateCount = newContentStates.get(previousState);
```

Comment:
Using HashMap.getOrDefault(*, 0) would lead to simpler code, here and further down.